drivers/d3d12/rendering_device_driver_d3d12.cpp

```diff
@@ -1858,22 +2090,219 @@ void RenderingDeviceDriverD3D12::command_buffer_end(CommandBufferID p_cmd_buffer
+		ERR_FAIL_COND_V(!SUCCEEDED(res), ERR_UNAVAILABLE);
+	} else {
+		swap_chain_desc.BufferCount = p_desired_framebuffer_count;
+		swap_chain_desc.Format = RD_TO_D3D12_FORMAT[swap_chain->data_format].general_format;
+		swap_chain_desc.BufferUsage = DXGI_USAGE_RENDER_TARGET_OUTPUT;
+		swap_chain_desc.SwapEffect = DXGI_SWAP_EFFECT_FLIP_DISCARD;
+		swap_chain_desc.SampleDesc.Count = 1;
+		swap_chain_desc.Flags = creation_flags;
+		swap_chain_desc.Scaling = DXGI_SCALING_NONE;
+
+		ComPtr<IDXGISwapChain1> swap_chain_1;
+		res = context_driver->dxgi_factory_get()->CreateSwapChainForHwnd(command_queue->d3d_queue.Get(), surface->hwnd, &swap_chain_desc, nullptr, nullptr, swap_chain_1.GetAddressOf());
+		ERR_FAIL_COND_V(!SUCCEEDED(res), ERR_CANT_CREATE);
+
+		swap_chain_1.As(&swap_chain->d3d_swap_chain);
+		ERR_FAIL_NULL_V(swap_chain->d3d_swap_chain, ERR_CANT_CREATE);
+
+		res = context_driver->dxgi_factory_get()->MakeWindowAssociation(surface->hwnd, DXGI_MWA_NO_ALT_ENTER | DXGI_MWA_NO_WINDOW_CHANGES);
+		ERR_FAIL_COND_V(!SUCCEEDED(res), ERR_CANT_CREATE);
+	}
+
+	res = swap_chain->d3d_swap_chain->GetDesc1(&swap_chain_desc);
+	ERR_FAIL_COND_V(!SUCCEEDED(res), ERR_CANT_CREATE);
+	ERR_FAIL_COND_V(swap_chain_desc.BufferCount == 0, ERR_CANT_CREATE);
+
+	surface->width = swap_chain_desc.Width;
+	surface->height = swap_chain_desc.Height;
+
+	swap_chain->creation_flags = creation_flags;
+	swap_chain->sync_interval = sync_interval;
+	swap_chain->present_flags = present_flags;
+
+	// Retrieve the render targets associated to the swap chain and recreate the framebuffers. The following code
+	// relies on the address of the elements remaining static when new elements are inserted, so the container must
+	// follow this restriction when reserving the right amount of elements beforehand.
+	swap_chain->render_targets.reserve(swap_chain_desc.BufferCount);
+	swap_chain->render_targets_info.reserve(swap_chain_desc.BufferCount);
+	swap_chain->framebuffers.reserve(swap_chain_desc.BufferCount);
+
+	for (uint32_t i = 0; i < swap_chain_desc.BufferCount; i++) {
+		// Retrieve the resource corresponding to the swap chain's buffer.
+		ID3D12Resource *render_target = nullptr;
+		res = swap_chain->d3d_swap_chain->GetBuffer(i, IID_PPV_ARGS(&render_target));
+		ERR_FAIL_COND_V(!SUCCEEDED(res), ERR_CANT_CREATE);
+		swap_chain->render_targets.push_back(render_target);
+
+		// Create texture information for the framebuffer to reference the resource. Since the states pointer must
+		// reference an address of the element itself, we must insert it first and then modify it.
+		swap_chain->render_targets_info.push_back(TextureInfo());
+		TextureInfo &texture_info = swap_chain->render_targets_info[i];
+		texture_info.owner_info.states.subresource_states.push_back(D3D12_RESOURCE_STATE_PRESENT);
+		texture_info.states_ptr = &texture_info.owner_info.states;
+		texture_info.format = swap_chain->data_format;
+		texture_info.desc = CD3DX12_RESOURCE_DESC(render_target->GetDesc());
```

Comment:
This breaks MinGW build.

Reply:
I checked your other PR, how strange that the headers literally define it differently. Now it's a bit late but maybe we have to make a wrapper around that function that works for both compilers without putting the compiler paths inside the caller methods.